### ISSUE TITLE
Update how vvl::Semaphore tracks timepoints

### ISSUE THREAD

**Issue by artem-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8369]
There is a difference between how core validation simulates forward progress as part of tracking resource `in-use` status and how syncval simulates queue forward progress.

*Core validation model* assumes that submission can be processed indefinitely slow - we can't make forward progress based on "enough time passed" or that we had many subsequent submits. Only after the host synchronization command (e.g. `vkWaitForFences`) we can advance the queue and update *in-use* status.

*Syncval submit validation* is different in a way that it does not care about host side in-use status and just needs to track how GPU command stream updates the resource, in this regard we process submissions as soon as we can (usually immediately during submission command) and don't need to wait for the host synchronization event, which can happen much later.

CoreVal and SyncVal use that same `vvl::Semaphore` state object and in some scenarios `vvl::Semaphore` can consume significant amount of memory that is not needed for syncval. Even bigger problem than resource usage is that it makes it more difficult to measure resource usage by syncval, we need to distinguish between allocations legally made by coreval and potential leaks in syncval. 

Ideas of possible solutions:
* Investigate if core validation *in-use* tracking actually needs to keep all timepoints, and if there is an opportunity to release them without affecting correctness of the algorithm.
* Configure `vvl::Semaphore` differently based validation object it is used with.

```
TEST_F(PositiveSyncValTimelineSemaphore, Test) {
    // TODO: existing implementation releases QBC objects and removes signals properly but vvl::Semaphore state
    // objects accumulate timepoints. Investigate if it's possible to safely release timepoint in vvl::Semaphore
    // or if should be special customization of vvl::Semaphore for syncval purposes, or maybe some other option.
    TEST_DESCRIPTION("This test helps to check that implementation releases SignalInfo and QueueBatchContexts objects");

    RETURN_IF_SKIP(InitTimelineSemaphore());

    const int N = 200'000;

    vkt::Buffer buffer_a(*m_device, 256, VK_BUFFER_USAGE_TRANSFER_SRC_BIT);
    vkt::Buffer buffer_b(*m_device, 256, VK_BUFFER_USAGE_TRANSFER_DST_BIT);
    m_command_buffer.begin(VK_COMMAND_BUFFER_USAGE_SIMULTANEOUS_USE_BIT);
    m_command_buffer.Copy(buffer_a, buffer_b);
    m_command_buffer.end();

    vkt::Semaphore semaphore(*m_device, VK_SEMAPHORE_TYPE_TIMELINE);
    for (int i = 1; i <= N; i++) {
        m_default_queue->Submit2WithTimelineSemaphore(m_command_buffer, semaphore, i - 1, semaphore, i);
    }
    m_device->Wait();
}
```
